Pipeline/PackageBuildOutputs/utilities/sbomGenerator.groovy

```diff
@@ -0,0 +1,206 @@
+    deployableArtifactComponentProperties.add(deployableArtifactContainerComponentProperty)
+    Property deployableArtifactDeployTypeComponentProperty = new Property()
+    deployableArtifactDeployTypeComponentProperty.setName("deployType")
+    deployableArtifactDeployTypeComponentProperty.setValue(deployableArtifact.deployType)
+    deployableArtifactComponentProperties.add(deployableArtifactDeployTypeComponentProperty)
+    deployableArtifactComponent.setProperties(deployableArtifactComponentProperties)
+    deployableArtifactComponent.setBomRef(newBomRef())
+    // Creates the dependencies for the Deployable Artifact
+    Dependency deployableArtifactDependency = new Dependency(deployableArtifactComponent.getBomRef())
+    ArrayList<Dependency> deployableArtifactDependencies = new ArrayList<Dependency>()
+    // Creates the list of sub-components
+    ArrayList<Component> deployableArtifactDependenciesComponents = new ArrayList<Component>()
+    // Add the source file itself
+    Component sourceFileComponent = new Component()
+    sourceFileComponent.setType(Component.Type.APPLICATION)
+	String sourceFile = record.getFile()
+    sourceFileComponent.setName(sourceFile)
+    sourceFileComponent.setBomRef(newBomRef())
+    ArrayList<Property> sourceFileProperties = new ArrayList<Property>()
+    Property sourceFileGitHashProperty = new Property()
+    sourceFileGitHashProperty.setName("gitHash")
+    sourceFileGitHashProperty.setValue(getGitHash(sourceFile, propertiesRecord))
+    sourceFileProperties.add(sourceFileGitHashProperty)
+    sourceFileComponent.setProperties(sourceFileProperties)
+    deployableArtifactDependenciesComponents.add(sourceFileComponent)
+    deployableArtifactDependencies.add(new Dependency(sourceFileComponent.getBomRef()))
+    // Now search for dependencies
+    if (dependencySetRecord) {
+        dependencySetRecord.getAllDependencies().each { physicalDependency ->
+            Component dependencyComponent = new Component()
+            dependencyComponent.setType(Component.Type.APPLICATION)
+            dependencyComponent.setName(physicalDependency.getLname())
+            ArrayList<Property> dependencyProperties = new ArrayList<Property>()
+            if (physicalDependency.getFile()) {
+                Property fileProperty = new Property()
+                fileProperty.setName("file")
+                fileProperty.setValue((physicalDependency.getSourceDir() != null ? physicalDependency.getSourceDir() : "") + physicalDependency.getFile())
+                dependencyProperties.add(fileProperty)
+                def dependencyGitHash = getGitHash(physicalDependency.getFile(), propertiesRecord)
+                if (dependencyGitHash != null && dependencyGitHash != "") {
+                    Property dependencyGitHashProperty = new Property()
+                    dependencyGitHashProperty.setName("gitHash")
+                    dependencyGitHashProperty.setValue(dependencyGitHash)
```

Comment:
The return value `null` will cause a NPE if the hash is not found. Can we handle the case?

Reply:
On line 139, it is supposed to retrieve the Git Hash first and go through that section only if it's different than `null`.